src/Microsoft.IdentityModel.Tokens/Validation/Validators.Algorithm.cs

```diff
@@ -42,5 +43,65 @@ public static void ValidateAlgorithm(string algorithm, SecurityKey securityKey,
                 });
             }
         }
+
+        /// <summary>
+        /// Validates a given algorithm for a <see cref="SecurityKey"/>.
+        /// </summary>
+        /// <param name="algorithm">The algorithm to be validated.</param>
+        /// <param name="securityKey">The <see cref="SecurityKey"/> that signed the <see cref="SecurityToken"/>.</param>
+        /// <param name="securityToken">The <see cref="SecurityToken"/> being validated.</param>
+        /// <param name="validationParameters"><see cref="TokenValidationParameters"/> required for validation.</param>
+        /// <param name="callContext"></param>
+#pragma warning disable CA1801 // TODO: remove pragma disable once callContext is used for logging
+        internal static AlgorithmValidationResult ValidateAlgorithm(string algorithm, SecurityKey securityKey, SecurityToken securityToken, TokenValidationParameters validationParameters, CallContext callContext)
+#pragma warning restore CA1801 // TODO: remove pragma disable once callContext is used for logging
+        {
+            if (validationParameters == null)
+            {
+                return new AlgorithmValidationResult(
+                    algorithm,
+                    ValidationFailureType.NullArgument,
+                    new ExceptionDetail(
+                        new MessageDetail(
+                            LogMessages.IDX10000,
+                            LogHelper.MarkAsNonPII(nameof(validationParameters))),
+                        typeof(ArgumentNullException),
+                        new StackFrame(true)));
+            }
+
+            if (validationParameters.AlgorithmValidator != null)
```

Comment:
Can we remove this? We are starting the process to remove all calls to delegates, as they will be implemented in the new ValidationParameters object and called from there.

Reply:
On the same note, we need to add the definition for the new delegate at the beginning of this file. I believe the name should be `AlgorithmValidatorDelegate` and match the signature to the new method returning the result type.

Reply:
Discussed offline, I'll make additional changes on a separate PR once, we have alignment on the overall design of the project.